common/parallel/TreeCrawler.go

```diff
@@ -80,8 +97,18 @@ func (c *crawler) start(ctx context.Context, root Directory) {
 	}
 	go heartbeat()
 
-	c.unstartedDirs = append(c.unstartedDirs, root)
+	if !c.isSource && c.tqueue != nil {
+		go c.receiveTqueue(root)
+	} else {
+		c.unstartedDirs = append(c.unstartedDirs, root)
+		go c.autoPacer()
+	}
+
 	c.runWorkersToCompletion(ctx)
+	if c.isSource && c.tqueue != nil {
+		close(c.closeAutoTune)
+		close(c.tqueue)
+	}
```

Comment:
Again, do it in the same form.. where we check for one cond and assert/panic for other.
Also, it's imp to add comments for such critical things.. f.e. here we should say something like.. now that  source traverser is done, close the tqueue channel to let target traverser know.